satpy/multiscene.py

```diff
@@ -46,18 +46,26 @@
 log = logging.getLogger(__name__)
 
 
-def stack(datasets, weights=None, combine_times=True):
-    """Overlay a series of datasets together.
+def stack(datasets, weights=None, combine_times=True, blend_type='select_with_weights'):
+    """Combine a series of datasets in different ways.
 
     By default, datasets are stacked on top of each other, so the last one applied is
-    on top. If a sequence of weights arrays are provided the datasets will
-    be combined according to those weights. The result will be a composite
-    dataset where the data in each pixel is coming from the dataset having the
-    highest weight.
+    on top. If a sequence of weights (with equal shape) is provided, the datasets will
+    be combined according to those weights. Datasets can be integers like 'ct', 'cma',
+    or radiances single channel or RGB composites. In the later case weights is applied
+    to each 'R', 'G', 'B' coordinate in the same way. The result will be a composite
+    dataset where each pixel is constructed in a way depending on variable blend_type.
```

Comment:
```suggestion
    be combined according to those weights. Datasets can be integer category products
    (ex. cloud type), single channels (ex. radiance), or RGB composites. In the
    latter case, weights is applied
    to each 'R', 'G', 'B' coordinate in the same way. The result will be a composite
    dataset where each pixel is constructed in a way depending on ``blend_type``.
```

I'd prefer if the documentation here avoided non-standard/reader-specific abbreviations like "ct" (I don't even know what "cma" is). I reworded your original paragraph, but I think it gets the same point across.

Could we also document the individual blend types here?

Note: I'm still reviewing everything here.

Reply:
@djhoese AFAIK 'cma' is cloud mask 'ctth' cloud top temperature height (?). Cloud type 'ct' is basically what Adam started with. I think the current documentation needs some overhaul as well. At least it was not easy to follow for me. Two things come to my mind: I propose that we do not use negative weights. If we allow for those Adam can simply set his weights to -satellite_zenith_angle  or to 90-satellite_zenith_angle which is satellite_elevation. If only positive weights are allowed Adam could use 180-satellite_zenith_angle. All these give the same results if the pixels are selected but these weights are more understandable than 1./satellite_zenith_angle. Sorry, for the missing tests I can probably be of no help. But I think it should be possible to start with Adam test as a basis.

Reply:
Ok on the abbreviations. My preference still stands. I'd prefer not to have docstrings refer to products that aren't self-explanatory or are reader-specific. Like `true_color` is obviously (at least hopefully for most Satpy users) expresses the general idea of an RGB composite.

Regarding negative weights, I haven't looked at the actual math, but does it currently (as written in this PR) work with negative weights or is that broken? What would change code-wise in multiscene.py if only positive weights were supported? What does this buy us (cleaner code? better performance?)? Would this then only require users (this is a brand new feature so I'm not worried about backwards compatibility much) to change how they generate weights? You mentioned it being more understandable than `1 / satellite_zenith_angle`, is this only used in the documentation for this weighting functionality or is it used somewhere else in Satpy?

Reply:
As I understand/assume 1./satellite_zenith_angle was just "invented" by Adam because he wanted a weight that decreases with increasing satellite_zenith_angle.  First I began with 1./(1.+ satellite_zenith_angle) to avoid division by zero. But finally I found out that any function that decreases with increasing satellite_zenith_angle leads to the same 'selected' result. No idea whether 1./satellite_zenith_angle is used elsewhere in satpy.

Reply:
Just tested weights = - satellite_zenith_angle with Adams selected version enhanced for bands: DOES NOT WORK :-). For blending negative weights this might lead to negative radiance or maybe byte underflow  (?). Certainly no option I think

![MetopB-20230301-DAY-0959-natural_color-eurol10-multi-blended](https://user-images.githubusercontent.com/101883219/222245701-3c322e44-8827-4fef-ab3c-29d9513211eb.jpg)
.

Reply:
The same three MetopB passes with weights = (180. - satellite_zenith_angle) work fine for blend_type='select_with_weights'.

![MetopB-20230301-DAY-0959-natural_color-eurol10-multi-blended](https://user-images.githubusercontent.com/101883219/222257889-1d6423ed-2cb0-46e7-a4c3-c2f75f9c285b.jpg)